unit/ansi-c/expr2c.cpp

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************\
+
+Module: Unit test for expr2c
+
+Author: Diffblue
+
+\*******************************************************************/
+
+#include <ansi-c/expr2c.h>
+#include <testing-utils/use_catch.h>
+#include <util/arith_tools.h>
+#include <util/bitvector_expr.h>
+#include <util/bitvector_types.h>
+#include <util/config.h>
+#include <util/namespace.h>
+#include <util/symbol_table.h>
+
+TEST_CASE("rol_2_c_conversion_unsigned", "[core][ansi-c][expr2c]")
+{
+  auto lhs = from_integer(31, unsignedbv_typet(32));
+  auto rhs = from_integer(3, unsignedbv_typet(32));
+  auto rol = shift_exprt(lhs, ID_rol, rhs);
+  CHECK(
+    expr2c(rol, namespacet{symbol_tablet{}}) ==
+    "31 << 3 % 32 | 31 >> 32 - 3 % 32");
+}
+
+TEST_CASE(
+  "rol_2_c_conversion_signed",
+  "[core][ansi-c][expr2c][convert_with_precedence]")
+{
+  config.ansi_c.mode = configt::ansi_ct::flavourt::GCC;
+  config.ansi_c.set_arch_spec_i386();
```

Comment:
Might be worth a comment explaining why this is necessary?

Reply:
Done